### PR TITLE
Bump to datafusion 9

### PR DESCRIPTION
**Pull request by timvw**

### CODE REVIEW

File: src/object_store/s3.rs [hunk 1]
```diff
@@ -120,9 +120,13 @@ impl S3FileSystem {
 #[async_trait]
 impl ObjectStore for S3FileSystem {
     async fn list_file(&self, uri: &str) -> Result<FileMetaStream> {
-        let (_, prefix) = uri.split_once("s3://").ok_or_else(|| {
-            std::io::Error::new(ErrorKind::Other, S3Error::AWS("No s3 scheme found".into()))
-        })?;
+
+        let prefix = if let Some((_scheme, path)) = uri.split_once("://") {
```
Comment (seddonm1) [inline src/object_store/s3.rs:123]: I wonder if there should be an assertion type check here for _scheme being s3?

File: src/object_store/s3.rs [hunk 2]
```diff
@@ -302,6 +306,7 @@ impl ObjectReader for AmazonS3FileReader {
 mod tests {
     use crate::object_store::s3::*;
     use aws_types::credentials::Credentials;
+    //use datafusion::arrow::record_batch::RecordBatch;
```
Comment (seddonm1) [inline src/object_store/s3.rs]: Remove if not being used

File: Cargo.toml [hunk 3]
```diff
@@ -25,13 +25,11 @@ aws-smithy-types = "0.39.0"
 aws-smithy-types-convert = { version = "0.39.0", features = ["convert-chrono"] }
 aws-types = "0.9.0"
 bytes = "1.1.0"
-datafusion-data-access = { version = "8.0.0" }
+datafusion = { version = "9.0.0" }
```
Comment (seddonm1) [inline Cargo.toml:28]: Its so hard to keep up with these API changes!

**Review by seddonm1 — CHANGES_REQUESTED** [review]
Wow this API moves fast but looks correct if tests are passing.

File: Cargo.toml (hunk 3 above)
Comment (timvw) [reply Cargo.toml:28]: Agreed. For now it's still manageable.

File: src/object_store/s3.rs (hunk 1 above)
Comment (timvw) [reply src/object_store/s3.rs:123]: I was inclined to agree with this, but now I'm less sure. Should we really care what the scheme is?

I can imagine use-cases where the user registers not only s3 but also s3a and/or s3n as scheme to be used with this ObjectStore implementation...

File: src/object_store/s3.rs (hunk 2 above)
Comment (timvw) [reply src/object_store/s3.rs]: Done

**Comment by timvw** [comment]
Not clear why tests fail on github actions. Works on my machine.

**Pull request closed without merging**